### ISSUE TITLE
String.toLowerCase and toUpperCase does not work with current locale

### ISSUE THREAD

**Issue by mkotsbak** [scala-js/scala-js#2551]
It seems like in SJS, the toUpperCase and toLowerCase are using JS's corresponding functions: (from RuntimeString.scala):

```
    def toLowerCase(): String = js.native
    def toUpperCase(): String = js.native
```

In JVM, those operations are working using the current locale. The similar behaviour in JS would correspond to using  these:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/toLocaleLowerCase
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/toLocaleUpperCase

**Comment by soc** [comment]
Please ping the scala-java-time project, because we use the current behavior. It's no big deal to move to calling JS' "real" `toLowerCase` instead, but we need to know it so we can fix it.

**Comment by andreaTP** [comment]
this seems to be pretty new (at least to me):

> Note: ECMAScript Internationalization API, implemented with Firefox 29, added the locales argument to the Number.toLocaleString() method. If the argument is undefined, this method returns localized digits specified by the OS

(from : [Number.toLocaleString](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toLocaleString#Example:_Checking_for_support_for_locales_and_options_arguments))

moreover it seems they are moving implementing this [specification](https://tools.ietf.org/html/rfc4647#section-3.4) as mentioned [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl#Locale_identification_and_negotiation) .

It does look promising!

**Comment by andreaTP** [comment]
Just to mention:
https://tc39.github.io/ecma402/#sec-defaultlocale

**Comment by mkotsbak** [comment]
@andreaTP that is great, but unfortunately as it is not possible to polyfill those features, it will take a long time before we can depend on them being available.

@sjrd is it possible somehow for an external dependency (like scala-java-locales) that knows the right locale to override these methods (as they are in RuntimeString)?

**Comment by sjrd** [comment]
It is unfortunately not possible to override methods defined in `RuntimeString`. I am prenuptial concrete proposals to include the necessary hooks in the library.

**Comment by gzm0** [comment]
I will close this as won't fix, until Scala.js as a whole has a clearer view on locale.

**Comment by gzm0** [comment]
See #1773 for overall locale tracking.
